test/integration/s3-aws.test.js

```diff
@@ -1,9 +1,102 @@
 const { S3 } = require('../../index')
+const { s3: s3AwsClient } = require('../../src/aws-client')
 
 const assert = require('assert').strict
+const path = require('path')
+const fs = require('fs')
+
+const BUCKET_NAME = 'bucket-test-integration'
+const FILE_PATH = path.join(__dirname, '/assets/index.csv')
+const FILE_KEY = 'index.csv'
+const LOCAL_PATH = '/data/tmp/'
 
 describe('Integration tests for s3', function () {
-  it('should upload file', function () {
-    assert.doesNotReject(S3.upload)
+  beforeEach(async function () {
+    await S3.createBucket(BUCKET_NAME)
+  })
+
+  afterEach(async function () {
+    await S3.deleteBucket(BUCKET_NAME)
+  })
+
+  it('should create bucket', async function () {
+    const bucketName = 'my-bucket-test-integration'
+    await assert.doesNotReject(S3.createBucket(bucketName))
+    const listBuckets = await s3AwsClient.listBuckets({}).promise()
+    const bucketsNames = listBuckets.Buckets.map((item) => item.Name)
+
+    assert(bucketsNames.includes(bucketName))
+
+    const params = {
+      Bucket: bucketName
+    }
+
+    await s3AwsClient.deleteBucket(params).promise()
+  })
+
+  it('should delete bucket', async function () {
+    const bucketName = 'my-bucket-test-integration'
+
+    const params = {
+      Bucket: bucketName
+    }
+
+    await s3AwsClient.createBucket(params).promise()
+
+    await assert.doesNotReject(S3.deleteBucket(bucketName))
+  })
+
+  it('should upload file', async function () {
+    await assert.doesNotReject(S3.uploadFile(BUCKET_NAME, FILE_PATH))
+
+    const files = await S3.listFilesInBucket(BUCKET_NAME)
+
+    assert(files.includes(FILE_KEY))
+
+    const params = { Bucket: BUCKET_NAME, Key: FILE_KEY }
+
+    await s3AwsClient.deleteObject(params).promise()
```

Comment:
É necessário esse trecho? No `afterEach` temos a deleção do Bucket `await S3.deleteBucket(BUCKET_NAME)` . Isso não exclui os arquivos juntos? Caso não exclua, não tem alguma forma de se remover todos os FILES_KEY no próprio `afterEach`

Reply:
Então, não pode deletar um bucket que contém arquivos.

Existe uma outra forma de criar um bucket mudando a policy dele:
https://stackoverflow.com/questions/62694166/how-to-force-delete-a-non-empty-s3-bucket-with-versioning-enabled

Mas como sairia muito fora do contexto da criação de um bucket comum, mantive a regra padrão dele que é deletar o bucket apenas quando está vazio.

Sobre colocar uma condição no `afterEach` verificando se tem arquivos e deletando eles, acredito que tenha, existe o `deleteObjects`, vou implementar aqui, boa (: